File: 1. Two Sum/solution.php
<?php 

    class Solution {
        /**
         * @param Integer[] $nums
         * @param Integer $target
         * @return Integer[]
         */
        function twoSum($nums, $target) {
            $n = count( $nums ) ; 
            $map = [ ] ;
            for( $i = 0 ; $i < $n ; ++$i  ){
                $curr = $target - $nums[$i] ; 
                if( isset( $map[$curr] ) ){
                    return [ $i , $map[ $curr ] ] ; 
                } 
                $map[ $nums[$i] ] = $i ; 
            }
            return [ ] ; 
        }
    }


?>

File: 1. Two Sum/solution.cpp
#include<iostream>
#include<vector>
#include<unordered_map>
using namespace std;
class Solution {
int n;
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        n = nums.size();
        unordered_map<int, int> mp;
        for (int i = 0; i < n; i++) {
            if (mp.find(target - nums[i]) != mp.end()) {
                return {mp[target - nums[i]], i};
            }
            mp[nums[i]] = i;
        }
        return {-1, -1};
    }
};